File: src/main/java/com/vc/easy/L53.scala
package com.vc.easy

object L53 {
  def maxSubArray(nums: Array[Int]): Int = {
    var i = 1
    val n = nums.length
    while(i < n) {
      nums(i) = Math.max(nums(i), nums(i) + nums(i - 1))
      i += 1
    }
    nums.max
  }
}

File: src/main/java/com/vc/easy/L53.java
package com.vc.easy;

class L53 {
    public int maxSubArray(int[] nums) {
        if(nums == null || nums.length == 0) return 0;

        int max = nums[0];
        for(int i = 1; i < nums.length; i++) {
            nums[i] = Math.max(nums[i], nums[i - 1] + nums[i]);
            max = Math.max(max, nums[i]);
        }
        return max;
    }
}
